src/pages/transactions/components/ListBox.tsx

```diff
@@ -0,0 +1,164 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { TransactionDetails } from './TransactionMatchDetails';
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import { useClientsQuery } from 'common/queries/clients';
+import { Client } from 'common/interfaces/client';
+import CommonProps from 'common/interfaces/common-props.interface';
+import { SearchArea } from './SearchArea';
+import { useTransactionMatchResourceQuery } from '../common/queries';
+import { ListBoxItem } from './ListBoxItem';
+
+export interface ResourceItem {
+  id: string;
+  name: string;
+  number: number;
+  clientName: string;
+  client_id: string;
+  status_id: string;
+  amount: number;
+  date: string;
+}
+
+export interface SearchInput {
+  searchTerm: string;
+  minAmount: number;
+  maxAmount: number;
+  startDate: string;
+  endDate: string;
+}
+
+interface Props extends CommonProps {
+  transactionDetails: TransactionDetails;
+  dataKey: 'invoices' | 'categories' | 'vendors';
+  setSelectedIds: Dispatch<SetStateAction<string[] | undefined>>;
+  selectedIds: string[] | undefined;
+}
+
+export function ListBox(props: Props) {
+  const [searchParams, setSearchParams] = useState<SearchInput>({
+    searchTerm: '',
+    minAmount: 0,
+    maxAmount: 0,
+    startDate: '',
+    endDate: '',
+  });
+
+  const [clientId, setClientId] = useState<string>('');
+
+  const { data: clientsResponse } = useClientsQuery();
+
+  const { data: resourceResponse } = useTransactionMatchResourceQuery(
+    props.dataKey,
+    searchParams.searchTerm,
+    clientId
+  );
+
+  const [resourceItems, setResourceItems] = useState<ResourceItem[]>();
+
+  const [clients, setClients] = useState<Client[]>();
+
+  const [isFilterModalOpened, setIsFilterModalOpened] =
+    useState<boolean>(false);
+
+  const [isInvoicesDataKey, setIsInvoicesDataKey] = useState<boolean>(false);
+
+  const isItemChecked = (itemId: string) => {
+    return Boolean(props.selectedIds?.find((id) => id === itemId)?.length);
+  };
+
+  const selectItem = (itemId: string, clientId?: string) => {
+    setClientId(clientId || '');
```

Comment:
This it not good way of handling something "that doesn't exists". Please make sure to use nullabe clientId.

Reply:
changed